src/process.ml

```diff
@@ -353,19 +353,27 @@ let run_internal ?dir ?(stdout_to=Output.stdout) ?(stderr_to=Output.stderr)
     end;
     n
   | WEXITED n ->
+    let show_command =
+      let error_starts_with_file = String.is_prefix output ~prefix:"File " in
+      Config.show_full_command_on_error ()
+      || (not error_starts_with_file)
+    in
     if display = Verbose then
       die "\n@{<kwd>Command@} [@{<id>%d@}] exited with code %d:\n\
            @{<prompt>$@} %s\n%s"
         id n
         (Colors.strip_colors_for_stderr command_line)
         (Colors.strip_colors_for_stderr output)
-    else
+    else if show_command then
       die "@{<error>%12s@} %a @{<error>(exit %d)@}\n\
            @{<details>%s@}\n\
            %s"
         progname Fancy.pp_purpose purpose n
         (Ansi_color.strip command_line)
         output
+    else
+      die "@{<error>%12s@} %a @{<error>(exit %d)@}\n%s"
```

Comment:
Should we even print this line? Unless we are using the `short` display mode, it seems superfluous

Reply:
If we remove it, we don't know which command failed. I think it's still helpful to know if it was ocamldep, ocamlc, etc.

Reply:
Ok, let's see how it goes. I was just mentioning this as inside Jane Street we've always shown the output and I've never found this to be an issue. It is only annoying with custom commands that don't show the location, however we've got this covered in this PR.

Reply:
Okay, then I will get rid of it then. If you have positive experience with this at JST, we might as well trust it.

Reply:
To be consistent, we should do the same when the commend exit with code 0 but still prints something, such as warnings.

Reply:
Okay, I did that but reused the same option. I'm wondering if we should rename the option now that it's not just applicable to errors.

Reply:
That seems reasonable yes, what about `--always-show-command-line`?